keps/sig-node/4033-group-driver-detection-over-cri/README.md

```diff
@@ -0,0 +1,880 @@
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+- [Infrastructure Needed (Optional)](#infrastructure-needed-optional)
+<!-- /toc -->
+
+## Release Signoff Checklist
+
+<!--
+**ACTION REQUIRED:** In order to merge code into a release, there must be an
+issue in [kubernetes/enhancements] referencing this KEP and targeting a release
+milestone **before the [Enhancement Freeze](https://git.k8s.io/sig-release/releases)
+of the targeted release**.
+
+For enhancements that make changes to code or processes/procedures in core
+Kubernetes—i.e., [kubernetes/kubernetes], we require the following Release
+Signoff checklist to be completed.
+
+Check these off as they are completed for the Release Team to track. These
+checklist items _must_ be updated for the enhancement to be released.
+-->
+
+Items marked with (R) are required *prior to targeting to a milestone / release*.
+
+- [ ] (R) Enhancement issue in release milestone, which links to KEP dir in [kubernetes/enhancements] (not the initial KEP PR)
+- [ ] (R) KEP approvers have approved the KEP status as `implementable`
+- [ ] (R) Design details are appropriately documented
+- [ ] (R) Test plan is in place, giving consideration to SIG Architecture and SIG Testing input (including test refactors)
+  - [ ] e2e Tests for all Beta API Operations (endpoints)
+  - [ ] (R) Ensure GA e2e tests meet requirements for [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md) 
+  - [ ] (R) Minimum Two Week Window for GA e2e tests to prove flake free
+- [ ] (R) Graduation criteria is in place
+  - [ ] (R) [all GA Endpoints](https://github.com/kubernetes/community/pull/1806) must be hit by [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md) 
+- [ ] (R) Production readiness review completed
+- [ ] (R) Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+This enhancement adds the ability for the container runtime to tell kubelet
+which cgroup driver to use. This removes the need for specifying cgroup driver
+in the kubelet configuration and eliminates the possibility of misaligned
+cgroup driver configuration between the kubelet and the runtime.
```

Comment:
Are we planning to support cgroupfs driver for cgroupsv2 as well? If not then by the time this KEP accepted and reached to GA, we might start deprecating cgroupsv1

I am not sure about timelines but this could be a possibility

Reply:
Are you talking about kubelet or the runtimes? I think kubelet at least supports cgroupfs for cgroupsv2, too(?)

Reply:
Worth considering that both the runtimes and kubelet have to be updated to land this in which time we might just only be doing systemd, @mrunalp @dchen1107 thoughts?

Reply:
I think that is a fair point we are moving towards a systemd cgroup driver default. Do we have a need to make something like this more general? Are there other features that kubelet would want to discover from the runtime?

Reply:
> maybe replaced with GC pinning on the CRI implementation side?

yeah the intention is this

in sig node we discussed this and also came up with QoS class, runtime class and CRI features (like CRI saying it supports user namespaces). However, for now, we opted to leave this KEP only focusing on cgroup driver

Reply:
Let's keep the possible future extensions in mind, as @mrunalp mentioned, keep it generalizable

Reply:
Seems like we can mark this thread resolved? :-)